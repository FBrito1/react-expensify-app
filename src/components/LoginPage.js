import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth'

export const LoginPage = ({ startLogin }) => (
    <div>
        <h1>Login Page</h1>
        <button  onClick={startLogin}>Log In</button>
    </div>
);

const MapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, MapDispatchToProps)(LoginPage);