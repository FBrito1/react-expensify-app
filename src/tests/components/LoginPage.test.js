import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { LoginPage } from '../../components/LoginPage';


test('should render Login page', () => {
    const wrapper = shallow(<LoginPage startLogin={() => {}} />)

    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should called startLogin on button click ', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} />);
    wrapper.find('button').simulate('click');

    expect(startLogin).toHaveBeenCalled();
});
