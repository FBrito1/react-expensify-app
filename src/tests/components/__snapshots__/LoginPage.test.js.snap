// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login page 1`] = `
<div>
  <h1>
    Login Page
  </h1>
  <button
    onClick={[Function]}
  >
    Log In
  </button>
</div>
`;
